flink-ml-iteration/src/test/java/org/apache/flink/iteration/operator/OperatorUtilsTest.java

```diff
@@ -0,0 +1,141 @@
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.api.operators.SimpleOperatorFactory;
+import org.apache.flink.streaming.api.operators.StreamOperatorFactory;
+import org.apache.flink.streaming.api.operators.TwoInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.tasks.StreamTaskMailboxTestHarnessBuilder;
+import org.apache.flink.streaming.runtime.tasks.TwoInputStreamTask;
+import org.apache.flink.util.OutputTag;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/** Tests the {@link OperatorUtils}. */
+public class OperatorUtilsTest extends TestLogger {
+
+    @Test
+    public void testCreateWrappedOperatorConfig() throws Exception {
+        StreamOperatorFactory<IterationRecord<Integer>> wrapperFactory =
+                new WrapperOperatorFactory<>(
+                        SimpleOperatorFactory.of(new MockTwoInputStreamOperator()),
+                        new AllRoundOperatorWrapper<>());
+
+        OperatorID operatorId = new OperatorID();
+
+        new StreamTaskMailboxTestHarnessBuilder<>(
+                        TwoInputStreamTask::new,
+                        new IterationRecordTypeInfo<>(BasicTypeInfo.STRING_TYPE_INFO))
+                .addInput(new IterationRecordTypeInfo<>(BasicTypeInfo.INT_TYPE_INFO))
+                .addInput(new IterationRecordTypeInfo<>(BasicTypeInfo.INT_TYPE_INFO))
+                .setupOutputForSingletonOperatorChain(wrapperFactory, operatorId)
+                .build();
+    }
+
+    @Test
+    public void testCreateWrappedOperatorConfigWithSideOutput() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.getConfig().enableObjectReuse();
+
+        final OutputTag<Integer> outputTag = new OutputTag("0", Types.INT) {};
+        DataStream<Integer> variableSource =
+                env.addSource(new DraftExecutionEnvironment.EmptySource<Integer>() {});
+
+        Iterations.iterateUnboundedStreams(
+                DataStreamList.of(variableSource),
+                DataStreamList.of(),
+                (variableStreams, dataStreams) -> {
+                    DataStream<Integer> variable = variableStreams.get(0);
+                    SingleOutputStreamOperator transformed =
+                            variable.transform(
+                                    "side-output",
+                                    Types.INT,
+                                    new MockSideOutputOperator<>(outputTag));
+                    return new IterationBodyResult(
+                            DataStreamList.of(variable),
+                            DataStreamList.of(transformed.getSideOutput(outputTag)));
+                });
+        env.execute();
```

Comment:
Can we additionally check the value of the output?

Since the test actually starts a job in mini-cluster, would it be simpler to add the test to an test class in flink-ml-tests?

Reply:
I added the check of the value of the output.

Reply:
`BroadcastOutputTest.java` uses `BroadcastOutput` directly and `OperatorEpochWatermarkTrackerFactoryTest.java` uses `OperatorEpochWatermarkTracker` directly. But `OperatorUtilsTest` does not use `OperatorUtilsTest` directly. Thus it seems better to move this test to e.g. `UnboundedStreamIterationITCase`.